contrib/src/main/java/macrobase/analysis/stats/KalmanFlattenedMatrixFilter.java

```diff
@@ -0,0 +1,23 @@
+package macrobase.analysis.stats;
+
+import macrobase.util.AlgebraUtils;
+import org.apache.commons.math3.linear.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Wrapper around KalmanVectorFilter, allows passing matrices to the Kalman Filter and receive matrices instead of vectors.
+ */
+public class KalmanFlattenedMatrixFilter extends KalmanVectorFilter {
```

Comment:
I like this api better, the filter has an internal state and operates on matrices or vectors or scalars alone. So it is clear which type of filter it is from the class.